src/core/location-hash.js

```diff
@@ -14,11 +14,12 @@ export function run(conf, doc, cb) {
   try {
     hash = decodeURIComponent(window.location.hash).substr(1);
   } catch (err) {
-    hash = "";
+    console.warn(err);
   }
   // Only scroll to the hash if the document hasn't been scrolled yet
   // this ensures that a page refresh maintains the scroll position
-  if (!hash && !document.documentElement.scrollTop) {
+  const { documentElement, body } = document;
+  if (!hash || documentElement.scrollTop || body.scrollTop) {
```

Comment:
Safari scrolls `body`, while other browsers scroll `documentElement`... the web 👍

Reply:
Also: we were doing it wrong earlier (logical error). The `||` was required whereas we were doing `&&`

Reply:
TIL `scrollingElement`